packages/nimbus/src/components/date-picker/date-picker.mdx

```diff
@@ -0,0 +1,169 @@
+- **Validate input:** Provide real-time validation and clear error messages for incorrect date formats or invalid dates.
+- **Show current date:** In a calendar picker, clearly highlight the current date for easy reference.
+- **Enable easy navigation:** Make it simple to navigate months and years within the calendar picker (e.g., using forward/backward buttons and year dropdown).
+- **Indicate selected date:** Clearly visually distinguish the selected date in the calendar.
+- **Handle unavailable dates:** If certain dates are unavailable (e.g., past dates for future events), clearly indicate them as disabled.
+- **Ensure keyboard accessibility:** Make all parts of the date field (input, calendar) navigable and operable using the keyboard.  
+- **Provide sufficient contrast:** Ensure adequate color contrast between text, icons, and background elements for accessibility.
+- **Localization:** Adapt the date format and calendar display to the user's locale.
+- **Offer preset ranges:** For date range selections, provide common presets like "Last week," "This month," etc.  
+
+### Usage
+
+Date field standardizes the creation of intuitive and accessible date inputs, ensuring consistent formatting, clear affordances for selection (like calendars), and robust accessibility for seamless user interaction across all date-related data capture. All uses should follow accessibility guides, ensuring color contrast, keyboard navigation, and clear messaging for invalid messages. 
+
+> [!TIP]\
+> When to use
+
+- **Specific date selection required:** When users need to choose a particular date, such as a birthdate, appointment date, or event date.
+- **Date-related tasks:** For features involving scheduling, filtering by date, setting deadlines, or defining date ranges.
+- **Standardized input format:** To ensure users enter dates in a consistent and predictable format, reducing data entry errors.
+- **Visual calendar navigation:** When users benefit from a visual representation of a calendar to navigate and select dates, especially for date ranges or when the day of the week is relevant.
+- **Accessibility needs:** To provide an accessible alternative to manual text input for users with disabilities, including keyboard navigation and screen reader support.
+- **Clear date context:** When the date is a primary piece of information and needs to be explicitly chosen.
+
+> [!CAUTION]\
+> When not to use
+
+- **Approximate or vague dates:** When the exact date is not critical, and users might only know a month or year (consider a select dropdown or a less specific input).
+- **Dates far in the past or future:** For dates like birthdates, consider a direct text input with clear formatting hints or segmented inputs (month/day/year) to avoid excessive calendar navigation. However, a date picker can still be suitable if well-implemented with year selection.
```

Comment:
I would consider our datepicker to be "well-implemented with year selection", and you can type the date into it directly without ever opening the picker, so this is less of a concern to me.  But the point is a good one so I'd guess it makes sense to have here?